contracts/cw4-group/src/contract.rs

```diff
@@ -309,7 +324,7 @@ mod tests {
         let err = update_members(
             deps.as_mut(),
             height + 5,
-            USER1.into(),
+            Addr::unchecked(USER1),
```

Comment:
Which would by the way also allow check `Copy`

Reply:
Did you see what I was doing with `AddrRef`?
It is `&str` to `Addr`'s `String`. This could be done with `const fn`

Reply:
We just need better back and forth between those types.

Reply:
`const fn` are executed at compile time and cannot do anything heap related. This is why you cannot have a Sting or Vec in a `const VARIABLE` . If we want to go that way, the address must be stored in a strict like

```rust
#[derive(Copy, ...)]
struct Addr {
    data: [u8; 128];
    length: usize;
}
```

However, this only gets handy if we can make a `const fn unchecked(input: &str) -> Addr` and I don't know if it is possible to copy the `&str` data into the array at compile time via other const functions.

Reply:
> We just need better back and forth between those types.

My thoughts exactly. `Into` calling `unchecked()` is a no no, as it would defeat `addr_validate`. Maybe a simple helper like `addr_unchecked()`, that lives in `cosmwasm_std` is the best we can ask for.

Another alternative is of course having a wrapper / marker type like `StringAddr` that implements `From<Addr>` using `unchecked`, but that is what we were trying to avoid when removing `HumanAddr`.

Reply:
I see all this concerns and we can tacle them one-by-one. Looking at the particular line I commented: Fewer conversions would be necessary if USER1 was an `Addr` instead of a `&str`. Converting `&str -> Addr` is hard because unsafe. But `Addr -> &str` is safe and easy. 

https://github.com/CosmWasm/cosmwasm/pull/869 shows an implementation that makes this possible.

Reply:
Nice. Will review tomorrow.